scripts/dotnet.py

```diff
@@ -532,6 +545,25 @@ def add_arguments(parser: ArgumentParser) -> ArgumentParser:
         help='{}'.format(CompilationAction.help_text())
     )
 
+    def __is_valid_sdk_version(version:str) -> str:
+        try:
+            if version is None or re.search('\d\.\d+\.\d+', version):
```

Comment:
What is the SDK version for preview builds?

Reply:
preview builds are \d.\d.\d+-preview#-\d+. This will match both preview and non-preview versions. It matches any string that contains \d.\d+\d+. We might want to harden it down a bit more so we don't match something like blah2.2.2blah.